frontend/src/Components/Game.jsx

```diff
@@ -88,6 +133,114 @@ class Game extends React.Component {
     }
   }
 
+  getAttack(event) {
+    let newKeyCode = event.keyCode;
+    if (newKeyCode === 69)
+      switch (this.state.direction) {
+        case "g":
+          if (this.state.xNPC === this.state.x - 1) {
+            this.setState({ NPCIsAlive: false });
+          }
+          break;
+        case "h":
+          if (this.state.yNPC === this.state.y - 1) {
+            this.setState({ NPCIsAlive: false });
+          }
+          break;
+        case "d":
+          if (this.state.xNPC === this.state.x + 1) {
+            this.setState({ NPCIsAlive: false });
+          }
+          break;
+        case "b":
+          if (this.state.yNPC === this.state.y + 1) {
+            this.setState({ NPCIsAlive: false });
+          }
+          break;
+      }
+  }
+
+  NPCmoves = [
+    "h",
+    "h",
+    "g",
+    "g",
+    "g",
+    "g",
+    "b",
+    "b",
+    "b",
+    "b",
+    "b",
+    "d",
+    "d",
+    "d",
+    "d",
+    "h",
+    "h",
+    "h"
+  ];
+
+  getNPCMove(indexNPCmove) {
```

Comment:
- Shouldn't change anything
- Code copy-paste, that's not a good aidea, couldn't you have done anything better ?